### PR TITLE
SASS-9722: Threshold between class 2 and 4

### PR DESCRIPTION
**Pull request by hs3t**

### CODE REVIEW

File: test/config/MessagesSpec.scala [hunk 1]
```diff
@@ -222,6 +222,7 @@ class MessagesSpec extends SpecBase {
     "checkNetProfitLoss.p1.profit.agent",
     "checkNetProfitLoss.p1.loss.agent",
     "whichYearIsLossReported.p2.loss.agent"
+//    "profitOrLossCalculation.class4Exempt.threshold"
```
Comment (hmrcvmu) [inline test/config/MessagesSpec.scala:225]: leftover?

File: app/controllers/journeys/adjustments/profitOrLoss/ProfitOrLossCalculationController.scala [hunk 2]
```diff
@@ -82,13 +84,19 @@ class ProfitOrLossCalculationController @Inject() (override val messagesApi: Mes
...
         case (true, _, true) => Some("tooOld")
         case _               => None
       }
     }
 
+  private def moreThanClass2ButLessThanClass4Threshold(taxYear: TaxYear, taxableProfitsAndLosses: List[TaxableProfitAndLoss])(implicit
+      request: DataRequest[_]): ApiResultT[Boolean] =
+    service.getUserDateOfBirth(request.nino, request.mtditid).map { _ =>
+      TaxableProfitAndLoss.class2EligibleButLessThenClass4Threshold(taxableProfitsAndLosses, taxYear)
+    }
```
Comment (Jake-Raffe) [inline app/controllers/journeys/adjustments/profitOrLoss/ProfitOrLossCalculationController.scala:98]: `service.getUserDateOfBirth` isn't being used so doesn't need to be ApiResultT.
Could we instead do
` def onPageLoad(taxYear: TaxYear, businessId: BusinessId): Action[AnyContent] = (identify andThen getData andThen requireData) async {
    implicit request =>
      val result = for {
        taxableProfitsAndLosses <- service.getAllBusinessesTaxableProfitAndLoss(taxYear, request.nino, request.mtditid)
        incomeSummary           <- service.getBusinessIncomeSourcesSummary(taxYear, request.nino, businessId, request.mtditid)
        netAmount          = incomeSummary.getNetBusinessProfitOrLossForTaxPurposes()
        profitOrLoss       = incomeSummary.returnProfitOrLoss()
        formattedNetAmount = formatSumMoneyNoNegative(List(netAmount))
        tables             = AdjustedTaxableProfitOrLossSummary.buildTables(taxYear, profitOrLoss)
        nicsExemptionMessage          <- showNicsExemptionMessage(taxYear, taxableProfitsAndLosses)
        showClass4AgeExemptionMessage <- showClass4AgeExemption(taxYear, taxableProfitsAndLosses)
      } yield Ok(
        view(
          request.userType,
          formattedNetAmount,
          taxYear,
          profitOrLoss,
          tables,
          nicsExemptionMessage,
          showClass4AgeExemptionMessage,
          journeys.routes.SectionCompletedStateController.onPageLoad(taxYear, businessId, ProfitOrLoss, NormalMode)
        )
      )
      handleResultT(result)
  }

  private def showNicsExemptionMessage(taxYear: TaxYear, taxableProfitsAndLosses: List[TaxableProfitAndLoss])(implicit
      request: DataRequest[_]): ApiResultT[Option[String]] =
    service.getUserDateOfBirth(request.nino, request.mtditid).map { userDoB =>
      val userIsClass2Eligible         = TaxableProfitAndLoss.areProfitsOrLossClass2Eligible(taxableProfitsAndLosses, taxYear)
      val ageIsTooYoung                = ageIsUnder16(userDoB, taxYear, ageAtStartOfTaxYear = false)
      val ageIsTooOld                  = !ageIsUnderStatePensionAge(userDoB, taxYear, ageAtStartOfTaxYear = false)
      val betweenEligibilityThresholds = TaxableProfitAndLoss.class2EligibleButLessThenClass4Threshold(taxableProfitsAndLosses, taxYear)

      (userIsClass2Eligible, ageIsTooYoung, ageIsTooOld, betweenEligibilityThresholds) match {
        case (true, true, _, _) => Some("class2Ineligible.tooYoung")
        case (true, _, true, _) => Some("class2Ineligible.tooOld")
        case (_, _, _, true)    => Some("betweenClass2AndClass4")
        case _                  => None
      }
    }`
    
    So we'd remove moreThanClass2ButLessThanClass4Threshold, and combine it into the pattern match

Comment (Jake-Raffe) [reply app/controllers/journeys/adjustments/profitOrLoss/ProfitOrLossCalculationController.scala:98]: Have dmed a formatted version

File: test/controllers/journeys/adjustments/profitOrLoss/ProfitOrLossCalculationControllerSpec.scala [hunk 3]
```diff
@@ -174,5 +178,43 @@ class ProfitOrLossCalculationControllerSpec extends ControllerSpec with TableDri
         assertEqualWithDiff(contentAsString(result), expectedView)
       }
     }
+
+    s"should show message when income more then class 2 but less then class 4 threshold" in {
+      val taxableProfit: BigDecimal = 10000
+      val taxableProfitsAndLosses   = List(TaxableProfitAndLoss(taxableProfit = taxableProfit, taxableLoss = BigDecimal(0)))
+      val stubService = SelfEmploymentServiceStub(
+        getAllBusinessesTaxableProfitAndLossResult = Right(taxableProfitsAndLosses),
+        getUserDateOfBirthResult = Right(adultDoB)
```
Comment (Jake-Raffe) [inline test/controllers/journeys/adjustments/profitOrLoss/ProfitOrLossCalculationControllerSpec.scala:187]: A valid DoB is the default response from the stub (already used in the 'should return OK and render correct view' of these tests) so can remove this line and `adultDoB `

**Review by hmrcvmu — APPROVED** [review]

**Pull request merged**